tests/MongoDB.Driver.Tests/IAggregateFluentExtensionsTests.cs

```diff
@@ -174,6 +174,17 @@ public void Group_should_generate_the_correct_document_using_expressions()
             AssertLast(subject, expectedGroup);
         }
 
+        [Fact]
+        public void Group_with_document_in_key_should_generate_the_correct_document_using_expressions()
+        {
+            var subject = CreateSubject()
+                .Group(x => x, g => new { Name = g.Select(x => x + " " + x.LastName).First() });
+
+            var expectedGroup = BsonDocument.Parse("{ $group : { _id : '$$ROOT', Name : { '$first' : { '$concat' : [ { '$add' : ['$$ROOT', ' '] }, '$LastName' ] } } } }");
```

Comment:
This looks wrong.

The `$add` operator can ONLY be used on numbers (or a date and a number) and `$$ROOT` is a DOCUMENT.